public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx

```diff
@@ -0,0 +1,650 @@
+            'medlineplus&v%3Asources=medlineplus-bundle&query='
+            + medlineTerm;
+        for (const concept of results.items) {
+            concept['synonymString'] = concept.conceptSynonyms.join(', ');
+            concept['drugBrands'] = concept.drugBrands;
+            if (domain.domain.toLowerCase() === 'measurement') {
+                concept.graphToShow = GraphType.Values;
+            } else {
+                concept.graphToShow = GraphType.BiologicalSex;
+            }
+        }
+        let standardConcepts = [];
+        if (results.standardConcepts) {
+            standardConcepts = results.standardConcepts;
+        }
+        let topResults = top10Results;
+        if (currentPage === 1) {
+            topResults = results.items.slice(0, 10);
+        }
+        this.setState({concepts: results.items, standardConcepts: standardConcepts, top10Results: topResults,
+        loading: false, medlineTerm: medlineTerm, medlinePlusLink: medlinePlusLink});
+  }
+
+
+   getTopConcepts() {
+        const {searchWord, domain: {domain}, measurementTestFilter, measurementOrderFilter} = this.state;
+        const searchRequest = {
+            query: searchWord,
+            domain: domain.toUpperCase(),
+            standardConceptFilter: StandardConceptFilter.STANDARDORCODEIDMATCH,
+            maxResults: 50,
+            minCount: 1,
+            pageNumber: 0,
+            measurementTests: measurementTestFilter ? 1 : 0,
+            measurementOrders: measurementOrderFilter ? 1 : 0
+        };
+        this.fetchConcepts(searchRequest);
+   }
+
+   fetchConcepts(searchRequest: any) {
+           dataBrowserApi().searchConcepts(searchRequest)
+               .then(results => {
+                    this.processSearchResults(results);
+               }).catch(e => {
+                       console.log(e, 'error');
+                       this.setState({loading: false});
+          });
+   }
+
+    handleChange(val) {
+        this.setState({ searchWord: val, loading: true, currentPage: 1, showTopConcepts: true});
+        this.domainTotals(val);
+    }
+
+    backToMain() {
+        if (this.state.searchWord) {
+                localStorage.setItem('searchText', this.state.searchWord);
+        }
+        navigateByUrl('');
+    }
+
+    selectConcept(concept: any) {
+        const {currentPage} = this.state;
+        if (concept && currentPage > 1) {
+                this.setState({selectedConcept: concept, currentPage: 1}, () => {this.getTopConcepts(); });
+        } else {
+                this.setState({selectedConcept: concept});
+        }
+    }
+
+    getTopResultsSize() {
+        const {top10Results, title} = this.state;
+        const top10ResultsTitle = (top10Results.length === 1) ? top10Results.length + ' ' + title.slice(0, -1) :
+        (top10Results.length < 10 ? top10Results.length + ' ' + title : 10 + ' ' + title);
+        return top10ResultsTitle;
```

Comment:
Nit: `top10ResultsTitle` var is redundant here, can just return the expression:
```suggestion
        return (top10Results.length === 1) ? top10Results.length + ' ' + title.slice(0, -1) :
        (top10Results.length < 10 ? top10Results.length + ' ' + title : 10 + ' ' + title);
```

Reply:
changed this thanks.